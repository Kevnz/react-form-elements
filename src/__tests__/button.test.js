import React from 'react'
import { render, fireEvent, cleanup } from 'react-testing-library'

import Button from '../button'

describe('TextBox component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { container } = render(<Button>Save</Button>)

      expect(container.firstChild).toMatchSnapshot()
    })
  })
  describe('handling event', () => {
    it('should set a value when passed in', () => {
      const mockHandler = jest.fn()
      const { container } = render(<Button onClick={mockHandler}>Save</Button>)
      expect(mockHandler.mock.calls.length).toBe(0)
      fireEvent.click(container.firstChild)
      expect(mockHandler.mock.calls.length).toBe(1)
    })
  })
})
