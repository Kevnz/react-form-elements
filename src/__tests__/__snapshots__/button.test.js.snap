// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox component The Rendered HTML should have a label 1`] = `
<button>
  Save
</button>
`;
